sklearn/conftest.py

```diff
@@ -265,6 +264,87 @@ def pyplot():
+    to a specific seed value while still being deterministic by default.
+
+    See the documentation for the SKLEARN_TESTS_GLOBAL_RANDOM_SEED
+    variable for instructions on how to use this fixture.
+
+    https://scikit-learn.org/dev/computing/parallelism.html#sklearn-tests-global-random-seed
+
+    """
+    # metafunc.config["random_seeds"] has been set in the
+    # handle_global_random_seed function based on
+    # SKLEARN_TESTS_GLOBAL_RANDOM_SEED
+    random_seeds = metafunc.config.getoption("random_seeds")
+
+    if "global_random_seed" in metafunc.fixturenames:
+        metafunc.parametrize("global_random_seed", random_seeds)
+
+
+def handle_global_random_seed(config):
+    """Part of pytest_configure that deals with the global_random_seed fixture."""
+    RANDOM_SEED_RANGE = list(range(100))  # All seeds in [0, 99] should be valid.
+    random_seed_var = environ.get("SKLEARN_TESTS_GLOBAL_RANDOM_SEED")
+
+    default_random_seeds = [42]
+
+    # When using pytest-xdist this function is called both in the xdist
+    # controller and xdist workers. In both cases, we rely on
+    # SKLEARN_TESTS_GLOBAL_RANDOM_SEED environment variable which is set in the
+    # main process and available in subprocesses
+    if random_seed_var is None:
+        random_seeds = default_random_seeds
+    elif random_seed_var == "all":
+        random_seeds = RANDOM_SEED_RANGE
+    else:
+        if "-" in random_seed_var:
+            start, stop = random_seed_var.split("-")
+            random_seeds = list(range(int(start), int(stop) + 1))
+        else:
+            random_seeds = [int(random_seed_var)]
+
+        if min(random_seeds) < 0 or max(random_seeds) > 99:
+            raise ValueError(
+                "The value(s) of the environment variable "
+                "SKLEARN_TESTS_GLOBAL_RANDOM_SEED must be in the range [0, 99] "
+                f"(or 'all'), got: {random_seed_var}"
+            )
+
+    # Set the random_seeds to that it can be accessed in other places e.g.
+    # pytest_generate_tests
+    config.option.random_seeds = random_seeds
+
+    class XDistHooks:
+        def pytest_configure_node(self, node) -> None:
+            # This passes the random seeds generated in the xdist controller to
+            # xdist workers
+            random_seeds = node.config.getoption("random_seeds")
+            node.workerinput["random_seeds"] = random_seeds
+
+    if config.pluginmanager.hasplugin("xdist"):
+        config.pluginmanager.register(XDistHooks())
+
+
+def pytest_report_header(config):
+    random_seed_var = environ.get("SKLEARN_TESTS_GLOBAL_RANDOM_SEED")
+    random_seeds = random_seed_var
+
+    return [
+        "To reproduce this test run, set the following environment variable:",
+        f'    SKLEARN_TESTS_GLOBAL_RANDOM_SEED="{random_seeds}"',
+        (
+            "See: https://scikit-learn.org/dev/computing/parallelism.html"
+            "#sklearn-tests-global-random-seed"
+        ),
+    ]
+
```

Comment:
With this PR, the header doesn't show up in the CI runs anymore. I think it has limited value since we always see the seed that was used in the pytest report of failed tests because it's in the parametrisation of the test. Also I don't think that any contributor has ever needed and even less seen this.

So I think it's fine to just remove this header. We can document the pytest command to run for a given seed in `parallelism.rst`.